platform/testing/end-to-end/vsp-cypress-resources.md

```diff
@@ -0,0 +1,835 @@
+
+#### initApplicationMock <a name="init-application-mock"></a>
+
+Import: `import { initApplicationMock } from './cypress-helpers'`
+
+Purpose: Sets up search results and a profile. Calling this also loads calculator_constants.
+
+Reference:
+
+```javascript
+export const initApplicationMock = (
+  profile = institutionProfile,
+  results = searchResults
+) => {
+  cy.route("GET", "/v0/gi/institutions/autocomplete**", autocomplete).as(
+    "defaultAutocomplete"
+  );
+  cy.route("GET", "/v0/gi/institutions/search**", results).as("defaultSearch");
+
+  initMockProfile(profile);
+};
+```
+
+#### hasFocusableCount <a name="has-focusable-count"></a>
+
+Import: `import { hasFocusableCount } from './cypress-helpers'`
+
+Purpose: Checks if count of focusable elements is correct. Focusable elements are those in the normal tab order (native focusable elements or those with tabIndex 0). The count logic will break on tabindexes > 0 because we do not want to override the browser's base tab order.
+
+Reference:
+
+```javascript
+export const hasFocusableCount = (selector, count) => {
+  const focusableElements = [
+    "a[href]",
+    "button",
+    "details",
+    'input[type="text"]',
+    'input[type="email"]',
+    'input[type="password"]',
+    'input[type="search"]',
+    'input[type="tel"]',
+    'input[type="url"]',
+    'input[type="radio"]',
+    'input[type="checkbox"]',
+    "select",
+    "textarea",
+    '[tabindex="0"]',
+    '[tabindex="-1"]',
+  ];
+  const msg = `Page does not contain ${count} focusable elements.`;
+  elementsWithinCount(focusableElements, selector, count, msg);
+};
+```
+
+#### hasTabbableCount <a name="has-tabbable-count"></a>
+
+Import: `import { hasFocusableCount } from './cypress-helpers'`
+
+Purpose: Checks if the count of tabbable elements is correct. Tabbable elements are those in the normal tab order (native focusable elements or those with tabIndex >= 0).
+
+Reference:
+
+```javascript
+export const hasTabbableCount = (selector, count) => {
+  const tabbableElements = [
+    "a[href]",
+    "button",
+    "details",
+    'input[type="text"]',
+    'input[type="email"]',
+    'input[type="password"]',
+    'input[type="search"]',
+    'input[type="tel"]',
+    'input[type="url"]',
+    'input[type="radio"]:checked',
+    'input[type="checkbox"]',
+    "select",
+    "textarea",
+    '[tabindex]:not([tabindex="-1"])',
+  ];
+
+  const msg = `Page does not contain ${count} tabbable elements.`;
+  elementsWithinCount(tabbableElements, selector, count, msg);
+};
+```
+
+#### initMockProfile <a name="init-mock-profile"></a>
+
+Import: `import { initMockProfile } from './cypress-helpers';`
+
+Purpose: Mocks the call for the profile.
+
+Reference:
+
+```javascript
+export const initMockProfile = (profile) => {
+  const facilityCode = profile.data.attributes.facility_code;
+  cy.route("GET", `/v0/gi/institutions/${facilityCode}`, profile).as(
```

Comment:
I'm guessing these might be API end points and this entire comment is off base. Please disregard if that's the case.

The heading for this section reads as though custom helpers are for the VAOS project only. This URL and several further down reference different projects, so I wasn't sure if the helpers are platform level (IE, available for any apps).